planterbox/feature.py

```diff
@@ -210,14 +210,21 @@ def check_scenarios(self):
                 scenario_steps,
                 scenario_examples,
             ) = scenario
-
             if scenario_examples:
                 # Do the example thing
                 unmatched = []
                 scenario_examples = list(self.load_examples(scenario_examples))
-                for scenario_example in scenario_examples:
-                    substituted_scenario = substitute_steps(scenario_steps, scenario_example)
-                    unmatched.extend(self.check_steps(substituted_scenario))
+                try:
+                    for scenario_example in scenario_examples:
+                        substituted_scenario = substitute_steps(scenario_steps, scenario_example)
+                        unmatched.extend(self.check_steps(substituted_scenario))
+                except UnmatchedSubstitutionException as ke:
```

Comment:
Do we think this should result in an `UnmatchedStepException` being raised at the end of the process? It does seem to fit with the intent of `check_scenarios`; in that there will be a step that isn't executable.